### PR TITLE
feat: add quick fix to disable check for line 

### PR DESCRIPTION
**Pull request by yassun7010**
Hi!

I'm interested in https://github.com/vscode-shellcheck/vscode-shellcheck/issues/116 and have created a PR to add the `addIgnoreComment` feature.

image

### CODE REVIEW

**Comment by felipecrs** [comment]
OMG this is really great! I will review soon.

File: src/linter.ts [hunk 1]
```diff
@@ -294,6 +305,31 @@ export default class ShellCheckProvider implements vscode.CodeActionProvider {
...
+        continue;
+      }
+
+      if (
+        typeof diagnostic.code === "object" &&
+        typeof diagnostic.code.value === "string" &&
+        diagnostic.code.value.startsWith("SC")
+      ) {
+        const ruleId = diagnostic.code.value;
+        const title = `ShellCheck: Add ignore comment for ${ruleId}`;
```
[the remark this answers is not in the document]
Comment (yassun7010) [reply src/linter.ts]: Of course! That is clearer.

File: src/linter.ts [hunk 2]
```diff
@@ -23,6 +23,7 @@ import {
 
 namespace CommandIds {
   export const runLint: string = "shellcheck.runLint";
+  export const addIgnoreComment: string = "shellcheck.addIgnoreComment";
```
Comment (felipecrs) [inline src/linter.ts:26]: We should probably update this too

```suggestion
  export const addIgnoreComment: string = "shellcheck.disableCheckForLine";
```

Or `shellcheck.disableForLine`

And maybe also rename the symbols, like variables and functions?

Comment (yassun7010) [reply src/linter.ts]: YES! I'll update the variables, etc.

Comment (yassun7010) [reply src/linter.ts]: Completed.

File: src/linter.ts [hunk 3]
```diff
@@ -411,6 +447,28 @@ export default class ShellCheckProvider implements vscode.CodeActionProvider {
...
+    ruleId: string,
+    range: vscode.Range,
+  ) {
+    if (!this.isAllowedTextDocument(textDocument)) {
+      return;
+    }
+    let position = new vscode.Position(
+      range.start.line,
+      textDocument.lineAt(range.start.line).range.start.character,
+    );
```
Comment (felipecrs) [inline src/linter.ts:461]: Again, thanks a lot for this PR. But before integrating it, I believe we need to at least come up with a way to make sure the newly added comment follows the proper indentation... example:


https://github.com/vscode-shellcheck/vscode-shellcheck/assets/29582865/dd909467-d552-4b16-a848-484ef3411121

Do you think you can handle this?

**Review by felipecrs — CHANGES_REQUESTED** [review]

File: src/linter.ts (hunk 3 above)
Comment (yassun7010) [reply src/linter.ts:462]: I am using `shfmt` and did not notice that.

I fixed it to take indentation into comment line!

**Review by felipecrs — APPROVED** [review]
LGTM

**Comment by felipecrs** [comment]
@yassun7010 thank you so much, this feature is really useful.

**Pull request merged**